src/io/flutter/logging/FlutterLogEntryParser.java

```diff
@@ -223,19 +249,57 @@ private static long timestamp(Event event) {
   }
 
   @Nullable
-  public FlutterLogEntry parse(@Nullable String id, @Nullable Event event) {
+  public List<FlutterLogEntry> parse(@Nullable String id, @Nullable Event event) {
     if (id != null && event != null) {
       switch (id) {
         case LOGGING_STREAM_ID:
           return parseLoggingEvent(event);
         case VmService.GC_STREAM_ID:
           return parseGCEvent(event);
+        case VmService.EXTENSION_STREAM_ID:
+          return parseExtensionEvent(event);
       }
     }
 
     return null;
   }
 
+  private List<FlutterLogEntry> parseExtensionEvent(@NotNull Event event) {
+    final JsonPrimitive extensionKind = event.getJson().getAsJsonPrimitive("extensionKind");
+    if (Objects.equals(extensionKind.getAsString(), "Flutter.Error")) {
+      return parseFlutterError(event.getJson());
+    }
+    return null;
+  }
+
+  private List<FlutterLogEntry> parseFlutterError(@NotNull JsonObject json) {
+    final List<FlutterLogEntry> entries = new ArrayList<>();
+
+    final JsonElement extensionData = json.get("extensionData");
+    final DiagnosticsNode diagnosticsNode = parseDiagnosticsNode(extensionData.getAsJsonObject());
+    if (diagnosticsNode != null) {
+      final String description = diagnosticsNode.toString();
+      final FlutterLogEntry entry = lineHandler.parseEntry(description, ERROR_CATEGORY, FlutterLog.Level.SEVERE.value);
+      // TODO(pq): process children: diagnosticsNode.getChildren().getNow(null) and set data to something structured.
+      entry.setData(json.toString());
+      entries.add(entry);
+    }
+    return entries;
+  }
+
+  @Nullable
+  private DiagnosticsNode parseDiagnosticsNode(@NotNull JsonObject json) {
+    if (inspectorObjectGroup == null) {
```

Comment:
how can this happen? Should this be an assert as it should generally indicate a logic bug.

Reply:
Updated.